src/entrypoints/my-redirect.ts

```diff
@@ -1,112 +1,95 @@
+  if (!redirectParams && !Object.keys(userParams).length) {
+    return "";
   }
-
-  protected render(): TemplateResult {
-    if (!this.redirect) {
-      return html``;
+  if (
+    Object.keys(redirectParams || {}).length !== Object.keys(userParams).length
+  ) {
+    throw Error("Wrong parameters");
+  }
+  Object.entries(redirectParams || {}).forEach(([key, type]) => {
+    if (!userParams[key] || !checkParamType(type, userParams[key])) {
+      throw Error("Wrong parameters");
     }
+  });
+  return `?${createSearchParam(userParams)}`;
+};
 
-    return html`
-      <my-instance-info
-        .instanceUrl=${this._instanceUrl}
-        @edit=${this._handleEdit}
-      ></my-instance-info>
+const render = () => {
+  const instanceUrl = getInstanceUrl();
 
-      <div class="card-actions">
-        <div></div>
-        <a href=${ifDefined(this._createRedirectUrl())}>
-          <mwc-button .disabled=${!this._instanceUrl}>Open Link</mwc-button>
-        </a>
-      </div>
-    `;
+  if (instanceUrl === null) {
+    setTimeout(() => document.location.assign("/?change=1"), 100);
+    return;
   }
 
-  private _handleEdit() {
-    document.location.assign("/?change=1");
+  let params;
+  try {
+    params = createRedirectParams();
+  } catch (err) {
+    alert("Invalid parameters given.");
+    document.location.assign(
+      `/create-link?redirect=${window.redirect.redirect}`
+    );
+    return;
   }
+  const redirectUrl = `${instanceUrl}/_my_redirect/${window.redirect.redirect}${params}`;
 
-  private _createRedirectUrl(): string | undefined {
-    if (!this._instanceUrl) {
-      return undefined;
-    }
-    const params = this._createRedirectParams();
-    return `${this._instanceUrl}/_my_redirect/${this.redirect}${params}`;
-  }
+  document.querySelector(".open-link")!.outerHTML = `
+    <a href="${redirectUrl}" class='open-link'>
+      <mwc-button>Open Link</mwc-button>
+    </a>
+  `;
 
-  private _createRedirectParams(): string {
-    const params = extractSearchParamsObject();
-    if (!this.params && !Object.keys(params).length) {
-      return "";
-    }
-    if (Object.keys(this.params || {}).length !== Object.keys(params).length) {
-      throw Error("Wrong parameters");
-    }
-    Object.entries(this.params || {}).forEach(([key, type]) => {
-      if (!params[key] || !this._checkParamType(type, params[key])) {
-        throw Error("Wrong parameters");
-      }
-    });
-    return `?${createSearchParam(params)}`;
-  }
+  const layout = document.querySelector(".layout")!;
 
-  private _checkParamType(type: ParamType, value: string) {
-    if (type === "string") {
-      return true;
-    }
-    if (type === "url") {
-      return value && value === sanitizeUrl(value);
-    }
-    return false;
+  let changeInstanceEl = document.querySelector(".instance-footer");
+  if (!changeInstanceEl) {
+    changeInstanceEl = document.createElement("div");
+    changeInstanceEl.classList.add("instance-footer");
```

Comment:
Why don't we add this to `redirect-page.html`?

Reply:
Because that page can only insert content into the ha-card

Reply:
🤦

Reply:
There is probably some eleventy magic I'm missing :)